src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosBlockSignatureRule.cs

```diff
@@ -86,13 +89,20 @@ private bool CheckBlockSignature(PosBlock block)
                 return false;
             }
 
-            if (ops.Count < 2) // script.GetOp(pc, opcode, vchPushValue)
+            if (ops.Count != 2)
             {
-                this.Logger.LogTrace("(-)[NO_SECOND_OP]:false");
+                this.Logger.LogTrace("(-)[INVALID_OP_COUNT]:false");
                 return false;
             }
 
             byte[] data = ops.ElementAt(1).PushData;
+
+            if (data.Length > MaxPushDataSize)
```

Comment:
Can't find any evidence of this limitation in the C++ code.

Reply:
script.cpp:

![image](https://user-images.githubusercontent.com/29645989/44697091-9f1fc380-aabd-11e8-8179-027e325dbc3e.png)

Reply:
script.h
![image](https://user-images.githubusercontent.com/29645989/44697176-ec9c3080-aabd-11e8-8454-d852b628e933.png)

Reply:
@bokobza I will call you today to explain